From cfcb84db5452b1fbc801ca85f2f70015660f3132 Mon Sep 17 00:00:00 2001
From: Lunar <lunar@anargeek.net>
Date: Wed, 6 Mar 2013 18:39:34 +0100
Subject: [PATCH] Make _open_subprocess argument more explicit in _handle_io

The previous way worked out, but was really harder to understand.
---
 gnupg.py |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/gnupg.py b/gnupg.py
index 4a73164..479e6dd 100644
--- a/gnupg.py
+++ b/gnupg.py
@@ -984,7 +984,11 @@ class GPG(object):
         """
         Handle a call to GPG - pass input data, collect output data.
         """
-        p = self._open_subprocess(args, passphrase is not None)
+        if passphrase is not None:
+            ask_passphrase = True
+        else:
+            ask_passphrase = False
+        p = self._open_subprocess(args, ask_passphrase)
         if not binary:
             stdin = codecs.getwriter(self.encoding)(p.stdin)
         else:
-- 
1.7.2.5

